File: Distributed Systems Project/BackEnd/immatriculation-service/src/main/java/fsm/miaad/immatriculationservice/entities/Vehicle.java
package fsm.miaad.immatriculationservice.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.persistence.*;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;


@Entity
@Data @AllArgsConstructor @NoArgsConstructor @ToString
@XmlAccessorType(XmlAccessType.FIELD)
@XmlRootElement
@Builder
public class Vehicle {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String licensePlate;
    private String brand;
    private int fiscalPower;
    private String model;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ManyToOne
    private Owner owner;
}

File: Distributed Systems Project/FrontEnd/radars-management-client-app/src/app/models/vehicle.ts
import {Owner} from "./owner";

export class Vehicle {
  id!:number;
  licensePlate!:string;
  brand!:string;
  fiscalPower!:number;
  model!:string;
  owner!:Owner;

  _links!: {
    self: {
      href: string;
    };
    vehicle: {
      href: string;
    };
    owner: {
      href: string;
    };
  };



}
